src/process.cpp

```diff
@@ -35,31 +37,42 @@ void process_end(bp::child &proc, bp::group &proc_handle) {
   proc.wait();
 }
 
-int exe(const std::string &cmd, bp::environment &env, file_t &file, std::error_code &ec) {
+int exe_with_full_privs(const std::string &cmd, bp::environment &env, file_t &file, std::error_code &ec) {
   if(!file) {
     return bp::system(cmd, env, bp::std_out > bp::null, bp::std_err > bp::null, ec);
   }
 
   return bp::system(cmd, env, bp::std_out > file.get(), bp::std_err > file.get(), ec);
 }
 
-int proc_t::execute(int app_id) {
-  if(!running() && _app_id != -1) {
-    // previous process exited on its own, reset _process_handle
-    _process_handle = bp::group();
+boost::filesystem::path find_working_directory(const std::string &cmd, bp::environment &env) {
+  // Parse the raw command string into parts to get the actual command portion
+  std::vector<std::string> parts = boost::program_options::split_unix(cmd);
```

Comment:
The failure testcase (Notepad2) seems not to proceed beyond this function.

Reply:
I think that the `working_path` is not being detected correctly in other cases too.

e.g.: Command `D:\Software\CinebenchR20\Cinebench.exe` with no working path defined.

Unlike Notepad, this app will launch correctly (because it's *not* in the system path?), but if I add some debugging to the code, it seems that the working directory is being set to `c:/` instead of the correct path `D:\Software\CinebenchR20\`.

Reply:
Yep, the reason is dumb. I am using `split_unix()` which treats `\` as an escape character. Using `split_winmain()` instead should take care of it.

Reply:
Is it difficult to correct the path separator automatically if the user enters the wrong one?

Reply:
It's non-trivial to try to sort out what the user intended, especially given the variety of ways you can enclose paths with spaces. I don't think that's likely to be a big issue but we can revisit if it turns out to be.